third_party/move/move-vm/runtime/src/loader/script.rs

```diff
@@ -218,17 +226,31 @@ impl ScriptCache {
         }
     }
 
-    pub(crate) fn get(&self, hash: &ScriptHash) -> Option<Arc<Function>> {
-        self.scripts.get(hash).map(|script| script.entry_point())
+    pub(crate) fn get(&self, hash: &ScriptHash) -> Option<&ScriptCacheEntry> {
+        self.scripts.get(hash)
     }
 
-    pub(crate) fn insert(&mut self, hash: ScriptHash, script: Script) -> Arc<Function> {
-        match self.get(&hash) {
-            Some(cached) => cached,
-            None => {
-                let script = self.scripts.insert(hash, script);
-                script.entry_point()
-            },
-        }
+    pub(crate) fn insert_compiled_script(
+        &mut self,
+        hash: ScriptHash,
+        compiled_script: CompiledScript,
+    ) -> Arc<CompiledScript> {
+        let compiled_script_to_return = Arc::new(compiled_script);
+        self.scripts.insert(
```

Comment:
What happens if we are inserting at a place that already contains a verified script? should that be allowed, no-op, or not expected from caller? If we introduce any logic let's also unit test.